### ISSUE TITLE
NPE with plugin

### ISSUE THREAD

**Issue by hgschmie** [HubSpot/dependency-management-maven-plugin#2]
Put the attached  [pom.xml.gz](https://github.com/HubSpot/dependency-management-maven-plugin/files/232537/pom.xml.gz) file into a project folder.  Run `mvn dependency-management:analyze`

```
[ERROR] Failed to execute goal com.hubspot.maven.plugins:dependency-management-maven-plugin:0.5:analyze (default-cli) on project deptest: Execution default-cli of goal com.hubspot.maven.plugins:dependency-management-maven-plugin:0.5:analyze failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.hubspot.maven.plugins:dependency-management-maven-plugin:0.5:analyze (default-cli) on project deptest: Execution default-cli of goal com.hubspot.maven.plugins:dependency-management-maven-plugin:0.5:analyze failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
    at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
    at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
    at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal com.hubspot.maven.plugins:dependency-management-maven-plugin:0.5:analyze failed.
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
    ... 20 more
Caused by: java.lang.NullPointerException
    at com.hubspot.maven.plugins.dependency.management.DependencyManagementAnalyzer.checkDependencyManagement(DependencyManagementAnalyzer.java:33)
    at com.hubspot.maven.plugins.dependency.management.DependencyManagementAnalyzer.analyze(DependencyManagementAnalyzer.java:25)
    at com.hubspot.maven.plugins.dependency.management.DependencyManagementMojo.execute(DependencyManagementMojo.java:33)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
    ... 21 more
```

**Comment by jhaber** [comment]
Derp, fixed by 20d0a09d247011b8b6827c6df20c04ff5c63f245 and released in version 0.6

**Comment by jhaber** [comment]
And in case the same thing can happen with plugins, I added 9623474bb3d8aaea9c61d8cfc590d8dabe6e6018 released in 0.7

**Comment by hgschmie** [comment]
Thanks! It may make sense to start a little set of integration tests. The duplicate-finder plugin (https://github.com/basepom/duplicate-finder-maven-plugin/) contains invoker based integration tests that helped a lot to lock down these corner cases.
